posix/netdb.c

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/netdb.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/sys/socket.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/posix/resolver.h>
+#include <openenclave/internal/trace.h>
+
+static size_t _resolver_table_len = 3;
+static oe_resolver_t* _resolver_table[3] = {0}; // At most 3
+
+/* Called by the public oe_load_module_hostresolver() function. */
+int oe_register_resolver(int resolver_priority, oe_resolver_t* presolver)
+{
+    int ret = -1;
+
+    if (presolver == NULL)
+        OE_RAISE_ERRNO(OE_EINVAL);
+
+    if (resolver_priority >= (int)_resolver_table_len)
+        OE_RAISE_ERRNO(OE_EINVAL);
+
```

Comment:
A lock here?

Reply:
Also perhaps in getaddrinfo and freeaddrinfo

Reply:
Thanks for catching this Radhika. I missed it during my review.